src/tools/loadbal.py

```diff
@@ -0,0 +1,517 @@
+*******************************************
+
+
+*  `qbicity(box_size,rc,skin)`:
+    
+    It's a function to check the system size cubicity, with a tolerance given by the rc+skin
+    `dLnorm` - (Lx,Ly,Lz)/Lmax 
+
+*  `changeIndex(dN,ima,imi)`:
+
+    It's a function that sorts the nodeGrid, according to the index of the maximum size of the 
+    box (ima) and its corresponding minimum (imi)
+    
+*  `nodeGridSizeCheck(node_gridX,node_gridY,node_gridZ)`:
+
+    It's a function that verifies if it is worthy to take care of special DD for inhomogeneous 	  
+    systems, otherwise the homogenous DD is triggered and returned as a xyz flags
+    nodeGridSizeCheck(node_gridX,node_gridY,node_gridZ)
+
+*  `halfDecomp(adrCenter1D,rc_skin,eh_size,halfCores1D,cellsX,ratioMS,idealGas)`:
+
+    It's a function that decomposes half of the box in one Dimension(1D) and it is assuming symmetry
+    in the initial simulation box. Even with irregularities in the DD it will find the half-Box-DD 
+
+*  `addHsymmetry(halfNeilListX,eh_size,rc_skin,node_gridX,cellsX,ratioMS,idealGas)`:
+    
+    Its a function that uses the previously decomposed half-box (from halfDecomp) and unfolds it to
+    match the whole Neighbors List. If it does not match the whole neighbor, due to any asymmetry in
+    the number of cores or the number of cells. It will be redistributed region by region by  
+    considering the whole simulation box size.
+
+*  `adaptNeiList(neiListxin)`:
+    
+    It's a function which adapts the number of cells that go into each core into the data structure of
+    left and right cell lists for example for 4 cores and 8 cells [3,4,5,8] to [0,3,4,5,8]
+
+*  `reDistCellsHom(node_gridX,sizeX,rc_skin)`:
+    
+    It's a function which distributes the cells into nodes as if they where homogeneous. It also
+    applies to inhomogeneous system whenever there are less than 2 cores per direction: X, Y or Z.
+
+*  `reDistCells(halfNeilListX,cellsX,eh_size,rc_skin,node_gridX,ratioMS,idealGas)`:
+    
+    It's a function which is matching proportion of cells to the cores on a dual resolution region
+    basis
+
+*  `redistDeltaRandomly(wholeNeiListX,deltaCells,totNodesEH=0,biased=0)`:
+    
+    It's a function which distributes the remaining DELTA cells into nodes semi-randomly. By default
+    the biase applies to the CG-region and it assumes ther cannot be more than 3 extra cells to
+    redistribute, because this is the total number of regions in the simulation box |CG|EH|CG| (by
+    default the cg biased is left this could be updated in the dyn load balancing case!
+
+*  `findNodesMS(node_gridX,totCellsEH,totCellsCG,ratioMS,idealGas)`:
+    
+    It's a function which normalizes the number of cells to go to the EH and CG regions and find the
+    ideal corresponding number of Nodes EH and CG
+
+"""
+
+from random import randint
+
+__author__ = 'Horacio V Guzman'
+__email__ = 'horacio.v.g at gmail dot com'
+__version__ = '1.0'
```

Comment:
Either make it it's own python package or don't version it separately that will just become very confusing for the users.

Reply:
Good Idea! it can run on its one... I am working on the package...